### PR TITLE
Keep element name when switching type

### LINKED ISSUE #1150: Keep element name when changing type
When you change the element type via the "cog dropdown", the element should retain the same name.

### PR DESCRIPTION
**Pull request by lisa-fehr**
Closes #1150 

Should only switch the name when changing type if it matches a default name for that type.

### CODE REVIEW

File: src/components/nodes/eventBasedGateway/index.js [hunk 1]
```diff
@@ -1,16 +1,19 @@
...
+const id = 'processmaker-modeler-event-based-gateway';
 
 export default {
-  id: 'processmaker-modeler-event-based-gateway',
+  id,
   component,
   bpmnType: 'bpmn:EventBasedGateway',
   control: false,
   category: 'BPMN',
-  label: 'Event-Based Gateway',
```
Comment (lisa-fehr) [inline src/components/nodes/eventBasedGateway/index.js]: This one was either `Event-Based Gateway` or `Event Based Gateway`. I switched it to the crown item.

File: tests/unit/keepOriginalName.spec.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+  });
+  it('Should switch default Gateway name', () => {
+    const node = new Node(
+      'processmaker-modeler-parallel-gateway',
+      { id: 'test', name: 'Exclusive Gateway' },
+      {},
+    );
+    expect(keepOriginalName(node)).false;
+  });
+});
```
[the remark this answers is not in the document]
Comment (lisa-fehr) [reply tests/unit/keepOriginalName.spec.js]: Might take a bit of time to set this one up.

File: src/components/nodes/node.js [hunk 3]
```diff
@@ -18,6 +26,26 @@ export default class Node {
     return this.type === type;
   }
 
+  isStartGroup() {
```
Comment (johnblackspear) [inline src/components/nodes/node.js]: Minor thing: I'm wondering if the concept of "group" will throw us later when we've forgotten the discussions around this. Would a clunkier but more specific name like `isStartEvent` or so work for you?

File: src/components/nodes/subProcess/index.js [hunk 4]
```diff
@@ -13,10 +14,10 @@ export default {
   control: false,
   category: 'BPMN',
   icon: require('@/assets/toolpanel/subProcess.svg'),
-  label: 'Sub Process',
+  label: defaultNames[id],
```
Comment (johnblackspear) [inline src/components/nodes/subProcess/index.js]: Just wondering - why is the label untranslated but the moddle name translated?

File: src/components/nodes/task/defaultNames.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+const defaultNames = {
+  'processmaker-modeler-task': 'Form Task',
```
Comment (johnblackspear) [inline src/components/nodes/task/defaultNames.js]: :D +1, this one eh

File: src/components/nodes/subProcess/index.js (hunk 4 above)
Comment (lisa-fehr) [reply src/components/nodes/subProcess/index.js]: `$t` isn't available here. We would need another solution to translate them.

**Comment by lisa-fehr** [comment]
Screen Shot 2020-03-30 at 2 52 47 PM
A solution for the edge case on 'Task with Marker' could be keeping the name if there are no default names.

**Pull request merged**